ConfigurationSubstitutor/RecursiveConfigVariableException.cs

```diff
@@ -0,0 +1,12 @@
+namespace ConfigurationSubstitution
+{
+    using System;
+
+    public class RecursiveConfigVariableException : Exception
```

Comment:
What do you think of EndlessRecursionVariableException ?
message could be : $"Variable {variableName} is causing an endless recursion"

Reply:
Done